adal/src/main/java/com/microsoft/aad/adal/BrokerAccountServiceHandler.java

```diff
@@ -152,14 +152,23 @@ public void onError(Throwable throwable) {
         final Throwable throwable = exception.getAndSet(null);
         //AuthenticationException with error code BROKER_AUTHENTICATOR_NOT_RESPONDING will be thrown if there is any exception thrown during binding the service.
         if (throwable != null) {
+            // Record this throwable to the BrokerEvent for reporting via telemetry
+            brokerEvent.setBrokerAccountServiceConnectionErrorInfo(throwable);
+
             if (throwable instanceof RemoteException) {
                 Logger.e(TAG, "Get error when trying to get token from broker: " + throwable.getMessage(), "", ADALError.BROKER_AUTHENTICATOR_NOT_RESPONDING, throwable);
+                brokerEvent.setBrokerError(BrokerEvent.BrokerError.BROKER_REMOTE_ERROR);
+                Telemetry.getInstance().stopEvent(brokerEvent.getTelemetryRequestId(), brokerEvent, EventStrings.BROKER_REQUEST_SILENT);
                 throw new AuthenticationException(ADALError.BROKER_AUTHENTICATOR_NOT_RESPONDING, throwable.getMessage(), throwable);
             } else if (throwable instanceof InterruptedException) {
                 Logger.e(TAG, "The broker account service binding call is interrupted. "  + throwable.getMessage(), "", ADALError.BROKER_AUTHENTICATOR_EXCEPTION, throwable);
+                brokerEvent.setBrokerError(BrokerEvent.BrokerError.BROKER_INTERRUPTED_ERROR);
+                Telemetry.getInstance().stopEvent(brokerEvent.getTelemetryRequestId(), brokerEvent, EventStrings.BROKER_REQUEST_SILENT);
                 throw new AuthenticationException(ADALError.BROKER_AUTHENTICATOR_NOT_RESPONDING, throwable.getMessage(), throwable);
             } else {
                 Logger.e(TAG, "Get error when trying to bind the broker account service." + throwable.getMessage(), "", ADALError.BROKER_AUTHENTICATOR_NOT_RESPONDING, throwable);
+                brokerEvent.setBrokerError(BrokerEvent.BrokerError.BROKER_BIND_ERROR);
```

Comment:
just leaving a note here, we are expecting bind error to be the major error right?

Reply:
maybe :P

---
In reply to: [168898983](https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/1122#discussion_r168898983) [](ancestors = 168898983)